src/host/CursorBlinker.cpp

```diff
@@ -9,22 +9,33 @@
 using namespace Microsoft::Console;
 using namespace Microsoft::Console::Interactivity;
 
+static void CALLBACK CursorTimerRoutineWrapper(_Inout_ PTP_CALLBACK_INSTANCE /*Instance*/, _Inout_opt_ PVOID /*Context*/, _Inout_ PTP_TIMER /*Timer*/)
+{
+    CONSOLE_INFORMATION& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
+
+    // There's a slight race condition here.
+    // CreateThreadpoolTimer callbacks may be scheduled even after they were canceled.
+    // But I'm not too concerned that this will lead to issues at the time of writing,
+    // as CursorBlinker is allocated as a static variable through the Globals class.
+    // It'd be nice to fix this, but realistically it'll likely not lead to issues.
+    gci.LockConsole();

```

Comment:
There's no TryLockConsole() anymore, so I replaced it with a worse solution. 😅
Using the new thread pool API we can simply tell the OS to not block us on exit. It's simpler so that's a win at least.

Reply:
This was really the only reason that `TryLockConsole` existed in the first place? Wow.